lib/yang.c

```diff
@@ -736,26 +736,20 @@ LY_ERR yang_parse_notification(LYD_FORMAT format, const char *data,
 		return err;
 	}
 
-	/*
-	 * Notification can be a child of some data node, so traverse the tree
-	 * until we find the notification.
-	 */
-	LYD_TREE_DFS_BEGIN (tree, dnode) {
-		if (dnode->schema->nodetype == LYS_NOTIF) {
-			found = true;
-			break;
-		}
-		LYD_TREE_DFS_END(tree, dnode);
+	err = lyd_find_xpath3(NULL, tree, xpath, NULL, &set);
+	if (err) {
+		zlog_err("Failed to parse notification: %s", ly_last_errmsg());
+		lyd_free_all(tree);
+		return err;
 	}
-
-	if (!found) {
-		zlog_err("Notification not found in the parsed tree");
+	if (set->count == 0) {
```

Comment:
Should it be `set->count != 1` instead? We don't expect more than one notification, at least until we start supporting notifications for datastore updates.

Reply:
I gave thought to this code when picking `== 0`. I chose to follow Postel's Law here. The code I'm replacing just returned the first found notification (whether that was the correct one or not). Here I'm doing better and returning what matches the actual xpath. If for some reasons we get back multiple matches (we shoulnd't) we don't break and still do something that is functionally correct.

Reply:
Right, ok.